### PR TITLE
Refactor wolfSSL_ASN1_TIME_adj to use GetFormattedTimeString (new API)

### PR DESCRIPTION
**Pull request by kaleb-himes**
# Description

After evaluating numerous approaches to resolving the ABI-check-test issues reported by GCC 9 on the cloud slaves @dgarske noticed that the code in wolfSSL_ASN1_TIME_adj in ssl.c was awfully similar (nearly a copy/paste) of some code from the function GetAsnTimeString in asn.c. To that end we decided to kill a few birds with 1 stone. This PR does a few things:

1) Adds a new internal API (not a public API) `GetFormattedTimeString` to return just the time string (not compatible with the existing function `GetAsnTime()`). The new function takes input tm structure, a buffer to hold the output and a length to control the XSNPRINTF number of bytes to be printed into that buffer.
2) Refactors `GetAsnTimeString` to call `GetFormattedTimeString ` (a new API for both use cases) before adding the ASN encoding onto the returned string
3) Refactors `wolfSSL_ASN1_TIME_adj` to use this same new API `GetFormattedTimeString `
4) Resolves the issues reported by the ABI-check-test 
5) Removes stack fluctuations between the UTC and GENERALIZE time cases by using a single buffer `uf_time` to store the formatted time when called from the original function in asn.c `GetAsnTimeString`

Fixes ABI-check-test

# Testing

Tested using the cloud.

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

File: wolfcrypt/src/asn.c [hunk 1]
```diff
@@ -12192,27 +12192,68 @@ int GetTimeString(byte* date, int format, char* buf, int len)
...
+    WOLFSSL_ENTER("GetAsnTimeString");
+
+    if (buf == NULL || len == 0)
+        return BAD_FUNC_ARG;
+
+    data_len = GetUnformattedTimeString(currTime, uf_time, len);
+    /* ensure room to add 2 bytes (ASN type and length) before proceeding */
+    if (len < data_len + 2)
+        return BUFFER_E;
+    if(data_len <= 0)
```
Comment (SparkiDev) [inline wolfcrypt/src/asn.c:12216]: Do this check before the length check.

Comment (SparkiDev) [reply wolfcrypt/src/asn.c:12216]: Formatting:
 if (data_len <= 0)
and:
 }
 else {

**Review by SparkiDev — CHANGES_REQUESTED** [review]

**Review by dgarske — APPROVED** [review]
I pushed a few cleanups. Great refactor!

**Review by kaleb-himes — COMMENTED** [review]
Thanks so much @dgarske for the welcome text and format cleanups!

- Changes welcome and Approved

**Review by SparkiDev — APPROVED** [review]

**Pull request merged**